nmostesting/suites/IS0502Test.py

```diff
@@ -593,3 +594,206 @@ def test_13(self, test):
+                                if "grain_rate" not in source:
+                                    return test.FAIL("No grain_rate found for Source {} associated with Sender {}"
+                                                     .format(source["id"], resource["id"]))
+                                if "grain_rate" in flow:
+                                    flow_rate = "{}/{}".format(flow["grain_rate"]["numerator"],
+                                                               flow["grain_rate"].get("denominator", 1))
+                                    if param_components[1] != flow_rate:
+                                        return test.FAIL("Exactframerate for Sender {} does not match its Flow {}"
+                                                         .format(resource["id"], flow["id"]))
+                                else:
+                                    source_rate = "{}/{}".format(source["grain_rate"]["numerator"],
+                                                                 source["grain_rate"].get("denominator", 1))
+                                    if param_components[1] != source_rate:
+                                        return test.FAIL("Exactframerate for Sender {} does not match its Source {} "
+                                                         "and is not overridden by the Flow"
+                                                         .format(resource["id"], source["id"]))
+                            elif param_components[0] == "TCS":
+                                if "transfer_characteristic" not in flow and param_components[1] != "SDR":
+                                    return test.FAIL("Transfer characteristic is missing from Flow attributes")
+                                elif "transfer_characteristic" in flow and \
+                                        flow["transfer_characteristic"] != param_components[1]:
+                                    return test.FAIL("TCS parameter for Sender {} does not match its Flow {}"
+                                                     .format(resource["id"], flow["id"]))
+                    elif fmtp and flow["media_type"] == "video/smpte291":
+                        for param in fmtp.group(1).split(";"):
+                            param_components = param.strip().split("=")
+                            if param_components[0] == "DID_SDID":
+                                did, sdid = param_components[1].strip("{").rstrip("}").split(",")
+                                if "DID_SDID" not in flow:
+                                    return test.FAIL("No DID_SDID found for Flow {} associated with Sender {}"
+                                                     .format(flow["id"], resource["id"]))
+                                found_match = False
+                                for did_sdid in flow["DID_SDID"]:
+                                    if did_sdid["DID"] == did and did_sdid["SDID"] == sdid:
```

Comment:
Should this handle the hex formatting vagaries of the two opposing specs identified in https://github.com/AMWA-TV/nmos-discovery-registration/issues/121? :disappointed:

Reply:
Technically yes